File: app/src/main/java/org/flyontime/android/ui/adapter/SearchViewFilterStrategy.java
package org.flyontime.android.ui.adapter;

import org.flyontime.android.model.data.ItemModel;

import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

/**
 * Created by JW on 06.05.2017.
 */

public class SearchViewFilterStrategy {

    public static List<ItemModel> getFilteredResults(List<ItemModel> mOriginalItems, String constraint) {
        List<ItemModel> results = new ArrayList<>();
        for (ItemModel item : mOriginalItems) {
            if (item.getCardTitle().toLowerCase(Locale.ROOT).contains(constraint.toLowerCase(Locale.ROOT))) {
                results.add(item);
            }
        }
        return results;
    }

}

File: app/src/main/java/org/flyontime/android/ui/adapter/SearchViewFilterStrategy.kt
package org.flyontime.android.ui.adapter

import org.flyontime.android.model.data.FlyOnTime.ItemModel
import java.util.*

/**
 * Created by JW on 06.05.2017.
 */

object SearchViewFilterStrategy {

    fun getFilteredResults(mOriginalItems: List<ItemModel>, constraint: String): List<ItemModel> {
        val results = ArrayList<ItemModel>()
        for (item in mOriginalItems) {
            if (item.cardTitle.toLowerCase(Locale.ROOT).contains(constraint.toLowerCase(Locale.ROOT))) {
                results.add(item)
            }
        }
        return results
    }

}
